src/escalus_connection.erl

```diff
@@ -239,7 +254,31 @@ get_stanza_safe(Client, Timeout, Pred) ->
                               continue
                       end
               end,
-    receive_stanzas(Client, Timeout, Handler).
+    Safe = maps:get(safe, Options, false),
+    WithMetadata = maps:get(with_metadata, Options, false),
+    Timeout = maps:get(timeout, Options, ?TIMEOUT),
```

Comment:
We could also redefine `TIMEOUT` macro as `default_timeout/0` function for better tracing.

Reply:
I can do it right now.